File: src/main/java/no/novari/flyt/configuration/validation/valueparsability/validators/BooleanParsabilityValidator.java
package no.novari.flyt.configuration.validation.valueparsability.validators;

import no.novari.flyt.configuration.model.configuration.entities.ValueMapping;
import no.novari.flyt.configuration.validation.valueparsability.ValueParsabilityValidator;
import org.springframework.stereotype.Service;

@Service
public class BooleanParsabilityValidator implements ValueParsabilityValidator {

    @Override
    public ValueMapping.Type getTypeToValidate() {
        return ValueMapping.Type.BOOLEAN;
    }

    @Override
    public boolean isValid(String value) {
        return value.equals("true") || value.equals("false");
    }

}

File: src/main/kotlin/no/novari/flyt/configuration/validation/valueparsability/validators/BooleanParsabilityValidator.kt
package no.novari.flyt.configuration.validation.valueparsability.validators

import no.novari.flyt.configuration.model.configuration.entities.ValueMapping
import no.novari.flyt.configuration.validation.valueparsability.ValueParsabilityValidator
import org.springframework.stereotype.Service

@Service
class BooleanParsabilityValidator : ValueParsabilityValidator {
    override fun getTypeToValidate(): ValueMapping.Type = ValueMapping.Type.BOOLEAN

    override fun isValid(value: String): Boolean = value == "true" || value == "false"
}
